### PR TITLE
Several little fixes for evading compile time errors

### PR DESCRIPTION
**Pull request by mkovero**
Hi, I'm in progress for compiling WLED with ESP-IDF (actually was able to do so), mostly due I would love to run it on ESP32-S2.
I stumbled on bunch of compile time errors due stricter nature of compiler settings on ESP-IDF.
I think some of these might be good to review through, I have not tested anything nor have any idea of previous development work regarding so results may vary.
Tyvm!

### CODE REVIEW

**Comment by srg74** [comment]
@mkovero Is your version of WLED really working with esp32s2? Do you mind to share your code?

File: wled00/e131.cpp [hunk 1]
```diff
@@ -180,7 +181,8 @@ void handleE131Packet(e131_packet_t* p, IPAddress clientIP, byte protocol){
         }
         uint16_t ledsTotal = previousLeds + (dmxChannels - dmxOffset +1) / 3;
         for (uint16_t i = previousLeds; i < ledsTotal; i++) {
-          setRealtimePixel(i, e131_data[dmxOffset++], e131_data[dmxOffset++], e131_data[dmxOffset++], 0);
+          setRealtimePixel(i, e131_data[dmxOffset], e131_data[dmxOffset], e131_data[dmxOffset], 0);
+          dmxOffset++;
```
Comment (not-jan) [inline wled00/e131.cpp:185]: This will only increment once instead of thrice

File: wled00/e131.cpp [hunk 2]
```diff
@@ -33,7 +33,8 @@ void handleDDPPacket(e131_packet_t* p) {
   realtimeLock(realtimeTimeoutMs, REALTIME_MODE_DDP);
   
   for (uint16_t i = offsetLeds; i < offsetLeds + packetLeds; i++) {
-    setRealtimePixel(i, data[c++], data[c++], data[c++], 0);
+    setRealtimePixel(i, data[c], data[c], data[c], 0);
+    c++;
```
Comment (not-jan) [inline wled00/e131.cpp:37]: This will only increment once instead of thrice

File: wled00/json.cpp [hunk 3]
```diff
@@ -124,7 +124,7 @@ void deserializeSegment(JsonObject elem, byte it)
           if (sz == 0 && sz > 4) break;
 
           int rgbw[] = {0,0,0,0};
-          byte cp = copyArray(icol, rgbw);
+         // byte cp = copyArray(icol, rgbw); Unused variable.
```
Comment (not-jan) [inline wled00/json.cpp:127]: Delete it then ;)

File: wled00/xml.cpp [hunk 4]
```diff
@@ -253,7 +253,7 @@ void getSettingsJS(byte subPage, char* dest)
     sappend('v',SET_F("LA"),strip.milliampsPerLed);
     if (strip.currentMilliamps)
     {
-      sappends('m',SET_F("(\"pow\")[0]"),"");
+      sappends('m',SET_F("(\"pow\")[0]"),std::string("").c_str());
```
Comment (not-jan) [inline wled00/xml.cpp:256]: This looks like an unnecessary heap allocation to me.

File: wled00/wled.cpp [hunk 5]
```diff
@@ -156,7 +156,7 @@ void WLED::setup()
   DEBUG_PRINT("esp8266 ");
   DEBUG_PRINTLN(ESP.getCoreVersion());
 #endif
-  int heapPreAlloc = ESP.getFreeHeap();
+ // int heapPreAlloc = ESP.getFreeHeap(); UNUSED
```
Comment (not-jan) [inline wled00/wled.cpp:159]: Can be removed then?

**Review by not-jan — COMMENTED** [review]
Some things I've noticed

**Comment by blazoncek** [comment]
Closing this PR due to the fact that WLED has progressed beyond the possibility to merge changes proposed.
I do not think it is sensible to keep this PR open for whatever reason. It is better to make a new one using the current code if there is a need to port WLED for ESP-IDF.

**Pull request closed without merging**